### PR TITLE
Mark Activity extension methods not required by spec as internal

### LINKED ISSUE #1544: Remove public extension methods on Activity not required for functionality/spec
Some methods like GetStatus,RecordException are by-design public APIs as it is serving spec needs.
There are a large number of extensions on Activity (eg: GetTag) which are special implementations to avoid allocations. These must not be made public.
For reuse within the repo, copy paste or use shared files.

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #1544 .

Need to do a follow up PR to possible split Benchmarks as it has ambigous ref to same extension method. Temporarily removed the file, to be added back next PR.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs [hunk 1]
```diff
@@ -203,30 +90,6 @@ public static void RecordException(this Activity activity, Exception ex)
...
-                {
-                    this.Value = item.Value;
-                    return false;
-                }
-
-                return true;
-            }
-        }
-
         private struct ActivityStatusTagEnumerator : IActivityEnumerator<KeyValuePair<string, object>>
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:93]: @cijothomas Did you mean to leave this here? Seems like it is also in the new file.

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs [hunk 2]
```diff
@@ -60,120 +60,7 @@ public static void SetStatus(this Activity activity, Status status)
...
-        /// <typeparam name="T">The struct <see cref="IActivityEnumerator{T}"/> implementation to use for the enumeration.</typeparam>
-        /// <param name="activityEvent">ActivityEvent instance.</param>
-        /// <param name="tagEnumerator">Tag enumerator.</param>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "ActivityProcessor is hot path")]
-        public static void EnumerateTags<T>(this ActivityEvent activityEvent, ref T tagEnumerator)
-            where T : struct, IActivityEnumerator<KeyValuePair<string, object>>
-        {
-            ActivityTagsEnumeratorFactory<T>.Enumerate(activityEvent, ref tagEnumerator);
+            return activity.GetStatusHelper();
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:63]: Might want to add `[MethodImpl(MethodImplOptions.AggressiveInlining)]`.

Comment (cijothomas) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:63]: just realized its already there, right?

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs]: missed this! thanks!

**Comment by cijothomas** [comment]
@CodeBlanch @reyang Please review. This will have cause a lot of merge conflicts with other PRs if left for a long time :D

File: src/OpenTelemetry.Api/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,10 @@
 ## Unreleased
 
 * Updated System.Diagnostics.DiagnosticSource to version 5.0.0
+* Mark Activity extension methods as internal as these are not required to be
+  public. GetTagValue,EnumerateTags, EnumerateLinks, EnumerateEvents. See
```
Comment (reyang) [inline src/OpenTelemetry.Api/CHANGELOG.md:7]:
```suggestion
  public. GetTagValue, EnumerateTags, EnumerateLinks, EnumerateEvents. See
```

**Review by eddynaka — APPROVED** [review]

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs (hunk 2 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:63]: Ah sorry it was above what the diff was showing.

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**